src/bindings/js/node/include/parameter_validation.hpp

```diff
@@ -0,0 +1,157 @@
+        return check_type(napi_boolean, key, value);
+    }
+
+    static void number(const std::string key, const Napi::Value& value) {
+        return check_type(napi_number, key, value);
+    }
+
+    static void string(const std::string key, const Napi::Value& value) {
+        return check_type(napi_string, key, value);
+    }
+
+    static void symbol(const std::string key, const Napi::Value& value) {
+        return check_type(napi_symbol, key, value);
+    }
+
+    static void object(const std::string key, const Napi::Value& value) {
+        return check_type(napi_object, key, value);
+    }
+
+    static void function(const std::string key, const Napi::Value& value) {
+        return check_type(napi_function, key, value);
+    }
+
+    static void bigint(const std::string key, const Napi::Value& value) {
+        return check_type(napi_bigint, key, value);
+    }
+
+    static void array(const std::string key, const Napi::Value& value) {
+        if (value.IsArray())
+            return;
+
+        std::string error_message = Signature::get_error_message(key, "Array", value.Type());
+        throw std::runtime_error(error_message);
+    }
+
+    static void check_type(const napi_valuetype expected_type, const std::string key, const Napi::Value& value) {
+        napi_valuetype real_type = value.Type();
+
+        if (real_type == expected_type)
+            return;
+
+        std::string error_message = Signature::get_error_message(key, expected_type, real_type);
+        throw std::runtime_error(error_message);
+    }
+
+    static const std::string get_error_message(std::string key,
+                                               napi_valuetype expected_type,
+                                               napi_valuetype real_type) {
+        std::string expected_type_str = get_type_name(expected_type);
+        std::string real_type_str = get_type_name(real_type);
+
+        return get_error_message(key, expected_type_str, real_type_str);
+    }
+
+    static const std::string get_error_message(std::string key, std::string expected, napi_valuetype real_type) {
+        std::string real_type_str = get_type_name(real_type);
+
+        return get_error_message(key, expected, real_type_str);
+    }
+
+    static std::string get_error_message(std::string key, std::string expected, std::string real) {
+        return "Argument #" + key + " has type '" + real + "', expected '" + expected + "'";
+    }
+};
+
+std::pair<bool, const std::string> validate_args(const Napi::CallbackInfo& info,
+                                                 const std::function<void(Signature&)>& builder) {
+    Signature s = Signature();
+    std::string validation_errors;
+
+    builder(s);
+
+    if (info.Length() != s.attributes_validators.size())
+        return std::make_pair(false, validation_errors);
+
+    size_t index = 0;
+
+    for (const auto& validator : s.attributes_validators) {
```

Comment:
Just for information as alternative, the index can have only loop scope.
```suggestion
    for (auto&& [index, validator] = std::make_pair(0, s.attributes_validators.begin()); validator != s.attributes_validators.end(); ++index; ++validator) {
```